src/js/BookReader.js

```diff
@@ -1397,24 +1412,45 @@ BookReader.prototype.jumpToIndex = function(index, pageX, pageY, noAnimate) {
 };
 
 /**
- * Switches the mode (eg 1up 2up thumb)
+ * Return mode or 1up if initial thumb
  * @param {number}
+ * @see BookReader.prototype.drawLeafsThumbnail
  */
-BookReader.prototype.switchMode = function(mode, options) {
-  if (mode === this.mode) {
-    return;
+BookReader.prototype.getPrevReadMode = function(mode) {
+  if (mode === BookReader.constMode1up || mode === BookReader.constMode2up) {
+    return mode;
+  } else if (this.prevReadMode === null) {
+    // Initial thumb, return 1up
+    return BookReader.constMode1up;
   }
+}
 
-  if (!this.canSwitchToMode(mode)) {
-    return;
+/**
+ * Switches the mode (eg 1up 2up thumb)
+ * @param {number}
+ * @param {object}
+ */
+BookReader.prototype.switchMode = function(mode, options = {}) {
+  // Set configuration, options overriding defaults
+  const {
+    init = false,
+    suppressFragmentChange = false,
+  } = options;
+
+  // Skip checks when init()
+  if (!init) {
```

Comment:
we may not need this `!init` check? perhaps we can always return "void" if `(mode === this.mode)` or `!this.canSwitchToMode(mode)`
it would be nice to find a way where `switchMode` has no concept of `init` - and just let it exist with certain rules.
not a show stopper, but if you think that it can be easily done, then i suggest we capture it with at least a ticket.

Reply:
No, nope. This is where the psghetti was in the init(). Because in the init() `mode` is always going to equal `this.mode` Before we were setting `this.mode` to `null`in init() after needed for other methods that needed it (but as a class var, not passed) in order to 'prime' this method. Bah!

I think that's psghetti. Things should be explicit and not sneaky magic. That said, sure, there might be a better way. At this point, (months and months,) way not worth it. And my brain would explode. But I'd start by refactoring using params and not class vars.
Thanks!

Reply:
>Things should be explicit and not sneaky magic.
+1.

yes, we can let this sit and iterate